### PR TITLE
Bump deps and use npm7

### PR DESCRIPTION
**Pull request by skjnldsv**
Please test on your respective apps and see what's breaking :warning: 
(list built automatically, feel free to remove unrelated entries)

1. Remove all devDeps like babel, eslint, webpack, loaders... etc
2. Delete your lockfile
3. Make sure you use npm 7 :warning: 
2. `npm i -D @nextcloud/webpack-vue-config@latest @nextcloud/eslint-config@latest @nextcloud/babel-config@latest @nextcloud/browserslist-config@latest @nextcloud/stylelint-config@latest`
3. Check your stylelint config https://github.com/nextcloud/stylelint-config#usage
4. Check your babel config  https://github.com/nextcloud/babel-config#usage
5. Check your eslint config  https://github.com/nextcloud/eslint-config#usage
6. Build and check for errors
7. Profit
_________
 - [ ] activity
 - [x] announcementcenter https://github.com/nextcloud/announcementcenter/pull/360
 - [x] app-tutorial https://github.com/nextcloud/app-tutorial/pull/319
 - [ ] approval
 - [ ] bookmarks
 - [ ] browser_warning
 - [ ] bruteforcesettings
 - [x] calendar https://github.com/nextcloud/calendar/pull/3215
 - [ ] cdav-library
 - [x] contacts https://github.com/nextcloud/contacts/pull/2281
 - [ ] cookbook
 - [x] deck https://github.com/nextcloud/deck/pull/3137
 - [ ] external
 - [ ] files_automatedtagging
 - [ ] files_pdfviewer
 - [ ] files_texteditor
 - [ ] files_zip
 - [ ] firstrunwizard
 - [ ] flow_notifications
 - [ ] flow_webhooks
 - [x] forms https://github.com/nextcloud/forms/pull/987
 - [ ] groupfolders
 - [ ] guests
 - [ ] integration_discourse
 - [ ] integration_github
 - [ ] integration_gitlab
 - [ ] integration_google
 - [ ] integration_jira
 - [ ] integration_mastodon
 - [ ] integration_moodle
 - [ ] integration_reddit
 - [ ] integration_twitter
 - [ ] integration_whiteboard
 - [ ] integration_zammad
 - [ ] mail
 - [ ] maps
 - [ ] nextcloud-password-confirmation
 - [ ] nextcloud-register
 - [ ] nextcloud-vue
 - [x] notes https://github.com/nextcloud/notes/pull/727
 - [ ] notifications
 - [ ] officeonline
 - [ ] password_policy
 - [x] photos https://github.com/nextcloud/photos/pull/801
 - [ ] polls
 - [ ] privacy
 - [ ] push
 - [x] registration https://github.com/nextcloud/registration/pull/309
 - [ ] richdocuments
 - [ ] ~~server :warning: will be the exception until Nextcloud 23 for safety reasons~~
 - [ ] sharelisting
 - [ ] social
 - [x] spreed https://github.com/nextcloud/spreed/pull/5773
 - [x] tasks https://github.com/nextcloud/tasks/pull/1638
 - [ ] terms_of_service
 - [ ] text
 - [ ] user_oidc
 - [x] user_retention https://github.com/nextcloud/user_retention/pull/361
 - [ ] viewer
 - [ ] workflow_pdf_converter
 - [ ] vue-dashboard

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -30,45 +30,19 @@
...
-		"sass-loader": "^8.0.2",
-		"stylelint": "^13.13.1",
-		"stylelint-config-recommended-scss": "^4.2.0",
-		"stylelint-scss": "^3.19.0",
-		"stylelint-webpack-plugin": "^2.1.1",
-		"url-loader": "^4.1.1",
-		"vue-loader": "^15.9.7",
-		"vue-template-compiler": "^2.6.13",
-		"webpack": "^4.46.0",
-		"webpack-cli": "^3.3.12"
```
Comment (nickvergessen) [inline package.json:72]: 👯👯👯👯👯👯👯👯👯👯👯👯👯👯👯

**Review by raimund-schluessler — APPROVED** [review]
Configuration works fine for Tasks.

**Review by korelstar — APPROVED** [review]
:tada: 
Works for notes, see https://github.com/nextcloud/notes/pull/727

**Review by jotoeri — APPROVED** [review]
LGTM https://github.com/nextcloud/forms/pull/987

**Comment by korelstar** [comment]
Just for information: if you use node v15, you do not need to explicitly install npm v7, since it's already included. This simplifies the GitHub workflow a little bit.

**Comment by skjnldsv** [comment]
> Just for information: if you use node v15, you do not need to explicitly install npm v7, since it's already included. This simplifies the GitHub workflow a little bit.

Node lts is still 14 :)
https://nodejs.org/en/about/releases/

**Comment by skjnldsv** [comment]
One issue I encountered!
If your vue is set to a different version than the vue-template-compiler of the webpack-vue-config, it will throw (well, I guess you will receive the update or will definitely see the issue). That's the only issue I found so far

**Review by marcoambrosini — APPROVED** [review]

**Comment by nickvergessen** [comment]
user retention works: https://github.com/nextcloud/user_retention/pull/361

**Comment by raimund-schluessler** [comment]
@skjnldsv I think nextcloud-vue is does not have a PR for npm7 yet, right (just asking because it's checked in the list)?

**Comment by skjnldsv** [comment]
> I think nextcloud-vue is does not have a PR for npm7 yet, right (just asking because it's checked in the list)?

Ah sorry, I thought https://github.com/nextcloud/nextcloud-vue/pull/2037 was it, indeed! :+1: 
Wanna do it there?

**Comment by raimund-schluessler** [comment]
> 
> 
> > I think nextcloud-vue is does not have a PR for npm7 yet, right (just asking because it's checked in the list)?
> 
> Ah sorry, I thought [nextcloud/nextcloud-vue#2037](https://github.com/nextcloud/nextcloud-vue/pull/2037) was it, indeed! 👍
> Wanna do it there?

Yes, I was about to try that yesterday already, but thought it would be cleaner to first merge the common webpack config 😃

**Comment by nickvergessen** [comment]
Registration works: https://github.com/nextcloud/registration/pull/309

**Review by juliushaertl — APPROVED** [review]

**Comment by skjnldsv** [comment]
Ok, let's go then, the process seems to be going well?

**Comment by skjnldsv** [comment]
@korelstar @raimund-schluessler @jotoeri 
You can adjust the package.json engine to also specify which npm version is needed:

```json
	"engines": {
		"node": ">=14.0.0",
		"npm": ">=7.0.0"
	},
```

**Comment by mejo-** [comment]
Collectives app also migrated now. Thanks a lot for the detailed instructions @skjnldsv!

Since in our case some extra steps were necessary (in order to make the CI work again), here's the link to our merge request for future reference: https://gitlab.com/collectivecloud/collectives/-/merge_requests/301

**Pull request merged**